net/uring/io_uring_linux.go

```diff
@@ -602,3 +602,35 @@ func (t *TUN) Close() error {
 	}
 	return err2
 }
+
+var (
+	mu sync.Mutex
+	// checks capabilities available on this system
+	capabilities        map[IORingOp]bool = map[IORingOp]bool{}
+	probingNotSupported bool              = false
+)
+
+// checkCapability returns whether probing is supported, and if and only if probing is
+// supported, whether the operation is also supported.
+// If probing is not supported, it may still be the case that the operation can be used.
+//
+// To test for a specific operation if there is no probing, probably running it and seeing if
+// the output is non-nil should be enough.
+func checkCapability(op IORingOp) (probingSupported bool, opOk bool) {
```

Comment:
I'm thinking about how to incorporate this into our current setup. What happens if probingSupported is false? Should we proceed or not? It appears that this style of probing isn't supported on the kernel I'm developing on (`5.4.0-66-generic`), and yet our code works. That says to me that probing this way isn't what we need, unfortunately. We might still want to do this in the future, to optionally use different ops, but our minimalist set appears to be enough.

So we need to find some different way to probe for io_uring support. Maybe we should start with making some unimportant io_uring syscall and checking whether it returns ENOSYS. Maybe later it'll turn out we also need to check the kernel version to require some bug fixes.

Reply:
For a specific syscall, probably it makes sense to just try it out and see if it fails either by returning null, or ENOSYS. That being said, if this is available this would be faster to check, so prefer probing, then populating the map with syscall attempts?

Reply:
Let's build the API to match our specific needs.

Our current need is to decide whether we can use io_uring; that's a binary decision. Probably the right way to make that decision--since probing is not available early enough--is to choose a single representative syscall and check for ENOSYS.